app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt

```diff
@@ -12,52 +12,68 @@ import com.github.chrisbanes.photoview.PhotoView
 import net.aiscope.gdd_app.ui.mask.BrushDiseaseStage
 
 @Suppress("TooManyFunctions")
+//TODO("Rename to PhotoMaskView")
 class MaskCustomView @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
 ) : PhotoView(context, attrs, defStyleAttr) {
 
     companion object {
-        const val MAX_SCALE = 5f
+        private const val MAX_SCALE = 5f
+        private const val MATRIX_SIZE = 9
     }
 
     enum class Mode {
         Zoom,
-        Draw
+        Draw,
+        Erase
     }
 
-    private val maskLayer: MaskLayer = MaskLayer(context, imageMatrix)
+    var onTouchActionUpListener: OnTouchListener? = null
+    private val maskLayerController = MaskLayer(context, imageMatrix)
```

Comment:
Because the prior name made me assume the object was actually a layer, which is incorrect... It controls something that acts like a layer (in a sense it draws something over what's been drawn), but is not actually a layer per se... Does it make sense? :-D

Reply:
Hmm, isn't the `MaskLayer` holding the... mask? 🥁 
Which makes it a layer? (think of photo editing software)
Or am I missing something?

Reply:
Na, let's take it back to maskLayer, don't worry. :-)
Thanks!